src/CommandLine/Xml/DiagnosticXmlSerializer.cs

```diff
@@ -90,6 +90,8 @@ private static XElement CreateSummary(IEnumerable<DiagnosticInfo> diagnostics, I
                     "Diagnostic",
                     new XAttribute("Id", f.Key.Id),
                     new XAttribute("Title", f.Key.Title.ToString(formatProvider)),
+                    new XAttribute("Description", f.Key.Description.ToString(formatProvider)),
+                    new XAttribute("HelpLink", f.Key.HelpLinkUri),
```

Comment:
1) I would change it from attribute to element because a) too many attributes make it less readable b) Description can be possibly long which make it unsuitable as an attribute value.

2) Both properties should be added to output only if they are not null.

Reply:
What about the existing fields (Title, Count). Should I refactor them to elements as well, or should I keep them as attributes for backward compatibility?

Reply:
I would keep it as they are because of backward compatibility.

Reply:
Okay, I made them elements. I also moved this functionality to a separate method.